sycl/test-e2e/WorkGroupMemory/test-plan.md

```diff
@@ -0,0 +1,148 @@
+# Test plan for [`sycl_ext_oneapi_work_group_memory`][spec-link] extension
+
+## Testing scope
+
+### Device coverage
+
+Functionality provided by the extension is not guarded by any aspects and
+should be supported on every device supported by an implementation.
+
+Therefore, those tests should be launched on every supported device
+configuration that we have.
+
+### Type coverage
+
+New APIs for allocating local memory described by the extension take data type
+as a template argument and therefore tests should be repeated for different
+data types to ensure that everything works correctly.
+
+List of data types which should be covered:
+- some basic C++ data types of different sizes: `char`, `int16_t`, `int`,
+  `double`
```

Comment:
Is `double` part of the list above or a separate type to be covered? Since `half` is mentioned by itself, I think it would make sense to do that for `double` too. Should we also mention that the tests for `double` and `half` require `aspect::fp64` and `aspect::fp16` respectively?

Reply:
> Is double part of the list above or a separate type to be covered? Since half is mentioned by itself, I think it would make sense to do that for double too. 

`half` is in a separate list, because it is not a basic C++ type, its a custom type introduced by SYCL specification.

I don't think that we should outline "optional" data types separately, because then we also need to consider `atomic_ref` for 64-bit types and I generally expect that the fact that certain tests have "extra" requirements and may  be skipped on certain HW as implied, simply because that is automatically the case if a test is properly written in accordance with the SYCL specification.